File: python/minilisp/stdenv.py
from minilisp import builtins
from minilisp import data


def make_top_level_env() -> data.Environment:
    env = data.Environment(parent=None)
    for name, func in builtins.ALL_MAP.items():
        env.ensure(name).value = data.Func(name, func)
    return env

File: javascript/minilisp/stdenv.js
const builtins = require('./builtins.js');
const data = require('./data.js');

function createTopLevelEnv() {
  const env = new data.Environment();
  for (const [name, value] of builtins.ALL.entries()) {
    env.ensure(name).value = value;
  }
  return env;
}

module.exports = {
  createTopLevelEnv,
};
